tests/flows/erc20source_multihop.go

```diff
@@ -0,0 +1,236 @@
+)
+
+func ERC20SourceMultihop(network interfaces.Network) {
+	cChainInfo := network.GetPrimaryNetworkInfo()
+	subnetAInfo, subnetBInfo := teleporterUtils.GetTwoSubnets(network)
+	fundedAddress, fundedKey := network.GetFundedAccountInfo()
+
+	ctx := context.Background()
+
+	// Deploy an ExampleERC20 on subnet A as the source token to be bridged
+	sourceTokenAddress, sourceToken := teleporterUtils.DeployExampleERC20(
+		ctx,
+		fundedKey,
+		cChainInfo,
+	)
+
+	// Create an ERC20Source for bridging the source token
+	erc20SourceAddress, erc20Source := utils.DeployERC20Source(
+		ctx,
+		fundedKey,
+		cChainInfo,
+		fundedAddress,
+		sourceTokenAddress,
+	)
+
+	// Token representation on subnet B will have same name, symbol, and decimals
+	tokenName, err := sourceToken.Name(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenSymbol, err := sourceToken.Symbol(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+	tokenDecimals, err := sourceToken.Decimals(&bind.CallOpts{})
+	Expect(err).Should(BeNil())
+
+	// Deploy an ERC20Destination for Subnet A
+	erc20DestinationAddress_A, erc20Destination_A := utils.DeployERC20Destination(
+		ctx,
+		fundedKey,
+		subnetAInfo,
+		fundedAddress,
+		cChainInfo.BlockchainID,
+		erc20SourceAddress,
+		tokenName,
+		tokenSymbol,
+		tokenDecimals,
+	)
+
+	// Deploy an ERC20Destination for Subnet B
+	erc20DestinationAddress_B, erc20Destination_B := utils.DeployERC20Destination(
+		ctx,
+		fundedKey,
+		subnetBInfo,
+		fundedAddress,
+		cChainInfo.BlockchainID,
+		erc20SourceAddress,
+		tokenName,
+		tokenSymbol,
+		tokenDecimals,
+	)
+
+	// Generate new recipient to receive bridged tokens
+	recipientKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	recipientAddress := crypto.PubkeyToAddress(recipientKey.PublicKey)
+
+	// Send tokens from C-chain to Subnet A
+	input := erc20source.SendTokensInput{
+		DestinationBlockchainID:  subnetAInfo.BlockchainID,
+		DestinationBridgeAddress: erc20DestinationAddress_A,
+		Recipient:                recipientAddress,
+		PrimaryFee:               big.NewInt(1e18),
+		SecondaryFee:             big.NewInt(0),
+		AllowedRelayerAddresses:  []common.Address{},
+	}
+	amount := big.NewInt(0).Mul(big.NewInt(1e18), big.NewInt(13))
+
+	receipt, bridgedAmount := utils.SendERC20Source(
+		ctx,
+		cChainInfo,
+		erc20Source,
+		erc20SourceAddress,
+		sourceToken,
+		input,
+		amount,
+		fundedKey,
+	)
+
+	// Relay the message to subnet A and check for ERC20Destination withdrawal
+	receipt = network.RelayMessage(
+		ctx,
+		receipt,
+		cChainInfo,
+		subnetAInfo,
+		true,
+	)
+
+	utils.CheckERC20DestinationWithdrawal(
+		ctx,
+		erc20Destination_A,
+		receipt,
+		recipientAddress,
+		bridgedAmount,
+	)
+
+	// Check that the recipient received the tokens
+	balance, err := erc20Destination_A.BalanceOf(&bind.CallOpts{}, recipientAddress)
+	Expect(err).Should(BeNil())
+	Expect(balance).Should(Equal(bridgedAmount))
+
+	// Multihop transfer to Subnet B
+	sendERC20MultihopAndVerify(
```

Comment:
Open to suggestions in terms of how to better abstract this between the 4 contracts. Unfortunately with the way bindings are generated with interfaces it doesn't make it as straightforward.